### ISSUE TITLE
[Enhancement] OneDrive Unterstützung für Konten auf microsoftonline.com

### ISSUE THREAD

**Issue by winnyschuster** [simatec/ioBroker.backitup#1128]
Die neue Version ist echt Klasse und so habe ich etwas rumgespielt und die oneDrive Option entdeckt. Leider hat sich da rausgestellt dass der Adapter 'nur' Konten von live.com unterstützt, nicht jedoch Schul- und Businesskonten, welche auf microsoftonline.com einloggen. D.h. das MS-Login erkennt erst gar nicht den Benutzernamen wenn man sich den Autorisierungscode holen möchte. 

Es wäre super wenn das irgendwann nachgerüstet werden könnte. Natürlich würde ich auch alpha-testen :-)

**Comment by simatec** [comment]
Kannst du mal bitte versuchen, den ersten Teil der in Backitup angezeigten URL ( bis zum Fragezeichen) mit folgender URL zu ersetzen?

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?

**Comment by winnyschuster** [comment]
Ja, jetzt kann ich mich einloggen und den Zugriff erteilen, allerdings wird der code nicht gespeichert. Im Log sehe ich einen Fehler bei GetRefreshToken mit StatusCode 400

**Comment by simatec** [comment]
Wir dir ein Code angezeigt in der Adresszeile?

**Comment by winnyschuster** [comment]
ja, ein code=xxxxxxxxxxxxxxx kommt dann, wenn ich alles hinter dem code=einfüge und auf `einreichen` klicke dann passiert scheinbar nichts ausser einem Refresh, der Button `Hole Code` wird wieder aktiv und die Meldung `OneDrive-Aktualisierungstoken:Nicht vorhanden` erscheint oder bleibt gleich.

Im LOG sehe ich: 
```
2024-06-15 23:53:01.123  - info: backitup.0 (20516) starting. Version 3.0.3 in /opt/iobroker/node_modules/iobroker.backitup, node: v20.14.0, js-controller: 6.0.4
2024-06-15 23:53:01.129  - debug: backitup.0 (20516) BackItUp has recognized a linux system
2024-06-15 23:53:01.145  - info: backitup.0 (20516) [iobroker] backup will be activated at 02:48 every 1 day(s)
2024-06-15 23:53:02.029  - warn: backitup.0 (20516) getToken Onedrive: AxiosError: Request failed with status code 400
2024-06-15 23:53:02.029  - warn: backitup.0 (20516) Onedrive Token: undefined
2024-06-15 23:53:04.031  - debug: backitup.0 (20516) No backup file was found
```

**Comment by simatec** [comment]
Ich prüfe das nochmal im Detail

**Comment by winnyschuster** [comment]
> Ich prüfe das nochmal im Detail

es gibt da auch unterschiedliche Anleitungen, für die neue Implentierung wäre [das wohl diese hier](https://learn.microsoft.com/de-de/onedrive/developer/rest-api/getting-started/aad-oauth?view=odsp-graph-online)

**Comment by simatec** [comment]
Kannst du bitte nochmal testen... Habe ein wenig was umgebaut.
Dafür müsstest du die aktuelle Github Version installieren und im Anschluss zur Sicherheit einen Upload machen

**Comment by winnyschuster** [comment]
wow! Alles funktioniert einwandfrei. Die Anmeldung funktioniert weiterhin für beide Accounts, und die Backups landen auf dem Account für welchen man zuletzt den entsprechenden Code eingereicht hatte. Der Wechsel ist etwas frickelig wenn man sich nicht vom Account abgemeldet hatte, dann erscheint die Auswahl nicht bei welchem Konto man sich anmelden will(nur bei erneuter Abfrage des Codes!!) Das hat aber **absolut nichts** mit **backitup** zu tun. Ich bin restlos begeistert! Herzlichen Dank für das super schnelle Umsetzen meines Wunsches, ich hoffe das nutzen mehr Leute als ich

**Comment by simatec** [comment]
Bitte teste nochmal die aktuelle Github Version. Ich musste die URL's nochmal anpassen, damit auch ältere Backitup Versionen weiterhin funktionieren.
Des Weiteren habe ich die Redirect URL angepasst und eine eigene gebaut, wo der Code besser dargestellt und kopiert werden kann.

**Comment by winnyschuster** [comment]
Funktioniert tadellos mit beiden Accounts und dass man den Code jetzt kopieren kann macht es noch einfacher.

**Comment by simatec** [comment]
Dann schließe ich das Issue hier als erledigt
